Solution 16

DDemoNZ opened the pull request.

DDemoNZ committed:
change get methods

DDemoNZ committed:
add new fields to User model (username, password, firstname, lastname)
add InjectInitializer
add servlet getAllUser and new Controllers

DDemoNZ committed:
add new fields to User model (username, password, firstname, lastname)
add InjectInitializer, Listener
add servlet getAllUser and new Controllers
add first variant of GetAllUsersController and jsp logic

DDemoNZ committed:
add new fields to User model (username, password, firstname, lastname)
add InjectInitializer, Listener
add servlet getAllUser and new Controllers
add first variant of GetAllUsersController and allUser.jsp logic
add first variant of RegistrationController and register.jsp file/logic

DDemoNZ committed:
added register form with adding user by username-psw and other attributes

DDemoNZ committed:
added register form with adding user by username-psw and other attributes
added adding items, remove items and getAll items servlets and jsp

DDemoNZ committed:
added register form with adding user by username-psw and other attributes
added adding items, remove items and getAll items servlets and jsp
add AddItemsToBucketController
add CompleteOrderController
add DeleteItemsFromBucketController
add orders.jsp and logic for user's orders page
improve main(internetShop) page
add bucket.jsp and logic

some improvements to other files

not finished GetUserOrdersController(!!!!!) and other tasks

DDemoNZ committed:
added other controllers and logic
added other jsp files
added css for jsp pages
fix and improve some methods in DAO and SERVICE layer (implementation and interface files)

optimized imports
and some other fixes

DDemoNZ committed:
some fix

DDemoNZ committed:
add "Main page" link to menu

DDemoNZ committed:
style for registration button

DDemoNZ committed:
some fix

DDemoNZ committed:
some fix

DDemoNZ committed:
some fix

DDemoNZ committed:
some fix

DDemoNZ committed:
some fix

Sofasmile commented on pom.xml:
```diff
@@ -20,11 +20,18 @@
             <artifactId>javax.servlet-api</artifactId>
             <version>3.1.0</version>
-            <scope>provided</scope>
+<!--            <scope>provided</scope>-->
```
давай видалимо цей коментар

Sofasmile commented on src/main/java/mate/academy/internetshop/App.java:
```diff
@@ -36,15 +36,15 @@
     }
 
     public static void main(String[] args) {
```
нам потрібен клас мейн?

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/AddItemsToBucketController.java:
```diff
@@ -0,0 +1,48 @@
+import mate.academy.internetshop.service.ItemService;
+import mate.academy.internetshop.service.UserService;
+
+public class AddItemsToBucketController extends HttpServlet {
```
```suggestion
public class AddItemToBucketController extends HttpServlet {
```

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/AddItemsToBucketController.java:
```diff
@@ -0,0 +1,48 @@
+    @Inject
+    private static UserService userService;
+
+    private static Long userId = 1L;
```
створи константу

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/AddItemsToBucketController.java:
```diff
@@ -0,0 +1,48 @@
+                .stream()
+                .filter(b -> b.getUserId().equals(user.getUserId()))
+                .findFirst()
+                .orElse(bucketService.create(new Bucket(user.getUserId())));
```
цю логіку потрібно винести в сервіс

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/DeleteOrderController.java:
```diff
@@ -0,0 +1,28 @@
+            throws ServletException, IOException {
+
+        String orderId = req.getParameter("order_id");
+        Order order = orderService.get(Long.valueOf(orderId));
```
нам потрібна ця стврока?

DDemoNZ replied:
ну в этом контроллере нет
я его убрал. Ошибка была
A в DelteUserOrdersController  там передаю в resp userId 
Т.е. передать именно тот userId который владелец этого заказа

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/DeleteUserOrdersController.java:
```diff
@@ -0,0 +1,29 @@
+import mate.academy.internetshop.model.Order;
+import mate.academy.internetshop.service.OrderService;
+
+public class DeleteUserOrdersController extends HttpServlet {
```
чим цей клас відрізняється від DeleteOrderController

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/GetAllOrdersController.java:
```diff
@@ -0,0 +1,29 @@
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        List<Order> orders = orderService.getAll();
```
мається на увазі всі ордери одного юзера

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/GetBucketController.java:
```diff
@@ -0,0 +1,40 @@
+    @Inject
+    private static BucketService bucketService;
+
+    private static Long userId = 1L;
```
константа

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/GetBucketController.java:
```diff
@@ -0,0 +1,40 @@
+                .stream()
+                .filter(b -> b.getUserId().equals(user.getUserId()))
+                .findFirst()
+                .orElse(bucketService.create(new Bucket(user.getUserId())));
```
в сервіс

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/GetUserOrdersController.java:
```diff
@@ -0,0 +1,34 @@
+    @Inject
+    private static UserService userService;
+
+    private static Long userId = 1L;
```
:)

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/InternetShopController.java:
```diff
@@ -0,0 +1,29 @@
+import mate.academy.internetshop.model.Item;
+import mate.academy.internetshop.service.ItemService;
+
+public class InternetShopController extends HttpServlet {
```
```suggestion
public class GetAllItemsController extends HttpServlet {
```

DDemoNZ replied:
а у меня получается GetAllItems есть и выводит список айтемов в страницу где их можно и добавлять и удалять
А этот выводит список айтемов в отдельную страницу где выбор "для юзера". Где можно их только добавлять в корзину.
Всё равно объединить ?

Sofasmile replied:
тоді давай переменуємо на GetAllItemsController(купівля) і GetAllItemsManageController(додавання, видалення)

Sofasmile commented on src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java:
```diff
@@ -22,25 +22,19 @@ public Bucket create(Bucket bucket) {
-            return updatedBucket;
-        }
-        return bucket;
+        Bucket oldBucket = get(bucket.getBucketId()).orElseThrow(()
```
можна спростити:
IntStream.range(0, Storage.buckets.size())
                .filter(x -> bucket.getBucketId().equals(Storage.buckets.get(x).getBucketId()))
                .forEach(i -> Storage.buckets.set(i, bucket));
        return bucket;

DDemoNZ replied:
Довольно запутанное упрощение чисто визуально)

Sofasmile replied:
ахахах) ви вирішували схожі задачки на стріми)

DDemoNZ committed:
remove Main
add "final" modifier for USER_ID field
moved some logic from controllers to service layer

DDemoNZ committed:
remove Main
add "final" modifier for USER_ID field
moved some logic from controllers to service layer
fix CompleteOrderController

DDemoNZ committed:
remove Main
add "final" modifier for USER_ID field
moved some logic from controllers to service layer
fix CompleteOrderController
improve update DAO methods

Sofasmile commented on src/main/java/mate/academy/internetshop/App.java:
```diff
@@ -1,14 +1,7 @@
-import mate.academy.internetshop.model.User;
 import mate.academy.internetshop.service.BucketService;
 import mate.academy.internetshop.service.ItemService;
 import mate.academy.internetshop.service.OrderService;
```
думаю що можна повністю видалити цей клас, або не пушити його в репозиторій

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java:
```diff
@@ -0,0 +1,45 @@
+    @Inject
+    private static OrderService orderService;
+
+    private static final Long USER_ID = 1L;
```
змінну бачу, але ж ти її не використовуєш) 
заміни bucketId

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java:
```diff
@@ -0,0 +1,45 @@
+        User user = userService.get(bucket.getUserId());
+        List<Item> items = bucket.getItems();
+        orderService.completeOrder(items, user);
+        bucketService.clear(bucket);
```
можливо цю логіку можна винести в сервіс?

DDemoNZ replied:
а что заносить? вроде и так в сервис кажется всё заносится
и используются команды сервисов

просто поиск юзера
это в сервис в метод "complete order" вставлять наверное не очень 

clean(bucket) возможно? 
но это как-то усложняет. Тут сразу clean сделал и передал нужный(текущий) бакет

передавать бакет возможно вместо List <Item из этого бакета в метод? 
единственные мысли по этому поводу
возможно если что-то подскажешь

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/AddItemToBucketController.java:
```diff
@@ -0,0 +1,43 @@
+        String itemId = req.getParameter("item_id");
+
+        Item item = itemService.get(Long.valueOf(itemId));
+        User user = userService.get(USER_ID);
```
ми десь використовуємо цей юзер?

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/GetAllOrdersController.java:
```diff
@@ -0,0 +1,38 @@
+            throws ServletException, IOException {
+
+        User user = userService.get(USER_ID);
+
```
чому через строку?

DDemoNZ replied:
не сразу конечно, но понял
ну изначально типа разные действия разделял 

там  с сервисами, потом req, потом resp 
исправить везде? Нельзя лишние строки совсем? Или можно в меру?

DDemoNZ committed:
del App
fix CompleteOrderController(bucket_id ---> USER_ID)

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java:
```diff
@@ -33,8 +33,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
 
         String bucketId = req.getParameter("bucket_id");
```
тоді і ця строка не потрібна

Sofasmile reviewed (commented):
а що стосовно решти коментарів?

DDemoNZ committed:
some fixes

DDemoNZ committed:
remove bucketService.clear() from CompleteOrderController
